grakn-engine/src/main/java/ai/grakn/engine/tasks/manager/singlequeue/SingleQueueTaskManager.java

```diff
@@ -142,8 +142,15 @@ public void addTask(TaskState taskState){
      * Stop a task from running.
      */
     @Override
-    public TaskManager stopTask(TaskId id, String requesterName) {
-        throw new UnsupportedOperationException("SingleQueueTaskManager does not support stopping tasks.");
+    public void stopTask(TaskId id, String requesterName) {
+        // TODO: Make only one call to storage if possible
+        if (!storage.containsTask(id)) {
+            // TODO: Figure out a nicer way than all these nulls...
+            TaskState task = TaskState.of(null, null, null, null, id).markStopped();
```

Comment:
Honestly, because I'm hoping there's a better way than using all these nulls. It doesn't seem right that all properties of a `TaskState` except ID can be null. For example, a `TaskState` until this point is meant to _always_ contain a class indicating the task to run, plus a schedule saying when to run it - now both of these things can be missing.